src/toplevel.c

```diff
@@ -165,6 +165,11 @@ jl_value_t *jl_eval_module_expr(jl_module_t *parent_module, jl_expr_t *ex)
     jl_module_t *newm = jl_new_module(name);
     jl_value_t *defaultdefs = NULL, *form = NULL;
     JL_GC_PUSH4(&last_module, &defaultdefs, &form, &newm);
+    // copy parent environment info into submodule
+    jl_sym_t *uuid_sym = jl_symbol("#uuid");
```

Comment:
would probably be better to make this part of the actual structure of the `jl_module_t`, rather than making it an cryptic error to define a function named `uuid`.

Reply:
It's named `#uuid` not `uuid` so you can't accidentally use this name; we can certainly do that once we get this working, but in the meantime, this was much easier to get going with.

Reply:
Functions do for their types, though — I believe this was Jameson's point:

```
julia> @eval $(Symbol("#uuid")) = 1
1

julia> @eval $(Symbol("#uuid"))
1

julia> uuid() = 2
uuid (generic function with 1 method)

julia> @eval $(Symbol("#uuid"))
typeof(uuid)
```

Reply:
Problem solved: https://github.com/JuliaLang/julia/pull/25455/commits/4fe702ee736731e6ffc308bbfc8fec3ec895c279. We can make it part of the module struct once this is all settled.

Reply:
Doesn't that conflict with mangled names in lowering now?

Reply:
I have no idea. How many hashes do I need to stick in front of this thing to make it safe?

Reply:
I think that depends on how many nested closures you want to support before it clashes?